### PR TITLE
Expose a function which reroots production (.closure.js) files.

### PR DESCRIPTION
**Pull request by mrmeku**
This function is exposed so that production bundling rules
(rollup, closure, unglify etc.) can invoke this function
to get a file tree contianing only production files with
the .closure.js suffix stripped off.

### CODE REVIEW

File: examples/es6_output/es6_output_test.sh [hunk 1]
```diff
@@ -2,15 +2,15 @@
 set -e
 
 # should not down-level ES2015 syntax, eg. `class`
-readonly GREETER_JS=$(cat $TEST_SRCDIR/build_bazel_rules_typescript/examples/es6_output/greeter.closure.js)
+readonly GREETER_JS=$(cat $TEST_SRCDIR/build_bazel_rules_typescript/examples/es6_output/es6_output.prod/examples/es6_output/greeter/greeter.js)
 if [[ "$GREETER_JS" != *"class Greeter"* ]]; then
   echo "Expected greeter.closure.js to contain 'class Greeter' but was"
```
Comment (alexeagle) [inline examples/es6_output/es6_output_test.sh]: update text

File: examples/es6_output/es6_output_test.sh [hunk 2]
```diff
@@ -2,15 +2,15 @@
...
   echo "$GREETER_JS"
   exit 1
 fi
 
 # should have native ES Module format
-readonly MAIN_JS=$(cat $TEST_SRCDIR/build_bazel_rules_typescript/examples/es6_output/main.closure.js)
+readonly MAIN_JS=$(cat $TEST_SRCDIR/build_bazel_rules_typescript/examples/es6_output/es6_output.prod/examples/es6_output/main.js)
 if [[ "$MAIN_JS" != *"import { Greeter }"* ]]; then
   echo "Expected main.closure.js to contain 'import { Greeter }' but was"
```
Comment (alexeagle) [inline examples/es6_output/es6_output_test.sh]: same

File: examples/es6_output/es6_output_test.sh [hunk 3]
```diff
@@ -2,15 +2,15 @@
 set -e
 
 # should not down-level ES2015 syntax, eg. `class`
-readonly GREETER_JS=$(cat $TEST_SRCDIR/build_bazel_rules_typescript/examples/es6_output/greeter.closure.js)
+readonly GREETER_JS=$(cat $TEST_SRCDIR/build_bazel_rules_typescript/examples/es6_output/es6_output.prod/examples/es6_output/greeter/greeter.js)
```
Comment (alexeagle) [inline examples/es6_output/es6_output_test.sh]: shouldn't this be `es6_output_lib.prod` ?

File: internal/tsc_wrapped/compiler_host.ts [hunk 4]
```diff
@@ -130,8 +130,8 @@ export class CompilerHost implements ts.CompilerHost, tsickle.TsickleHost {
 
   /** Allows suppressing warnings for specific known libraries */
   shouldIgnoreWarningsForPath(filePath: string): boolean {
-    return this.bazelOpts.ignoreWarningPaths.some(
-        p => !!filePath.match(new RegExp(p)));
+    return (this.bazelOpts.ignoreWarningPaths || [])
```
Comment (alexeagle) [inline internal/tsc_wrapped/compiler_host.ts]: is this change required for it to work?

File: internal/common/compilation.bzl [hunk 5]
```diff
@@ -292,6 +290,10 @@ def compile_ts(ctx,
   if not is_library:
     files += depset(tsickle_externs)
 
+  transitive_es6_srcs = depset()
+  for dep in ctx.attr.deps:
+    transitive_es6_srcs += getattr(dep.typescript, "transitive_es6_srcs", [])
```
Comment (alexeagle) [inline internal/common/compilation.bzl]: I think it would be better to do this with an aspect, maybe we should pair on that?

File: internal/common/reroot_prod_files.bzl [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  rerooted_es6_srcs = depset()
+
+  for dep in ctx.attr.deps:
+    if hasattr(dep, "typescript"):
+      for es6_src in dep.typescript.transitive_es6_srcs:
+        rerooted_es6_src = ctx.actions.declare_file(
+          "%s.prod/%s" % (
+            ctx.label.name,
+            es6_src.short_path.replace(".closure.js", ".js")))
+        ctx.actions.expand_template(
```
Comment (alexeagle) [inline internal/common/reroot_prod_files.bzl]: comment that this is a cheap way of doing a file copy?

File: examples/es6_output/es6_output_test.sh (hunk 3 above)
Comment (mrmeku) [reply examples/es6_output/es6_output_test.sh]: No, we use the label of the bundling rule rather than each individual ts_library since we are rerooting under a single tree rather than many.

File: examples/es6_output/es6_output_test.sh (hunk 1 above)
Comment (mrmeku) [reply examples/es6_output/es6_output_test.sh]: Donezo

File: examples/es6_output/es6_output_test.sh (hunk 2 above)
Comment (mrmeku) [reply examples/es6_output/es6_output_test.sh]: Donezo

File: internal/common/compilation.bzl (hunk 5 above)
Comment (mrmeku) [reply internal/common/compilation.bzl]: I did it myself! Watching you do it yesterday was all I needed

File: internal/tsc_wrapped/compiler_host.ts (hunk 4 above)
Comment (mrmeku) [reply internal/tsc_wrapped/compiler_host.ts]: No, but it is a bug in rules_typescript that we might as well merge in. I ran into it frequently while doing ABC work

Comment (alexeagle) [reply internal/tsc_wrapped/compiler_host.ts]: hmm, I'd rather keep it separate, and also understand why this is sometimes not set (I think it's easier to understand if the data is not nullable)

**Comment by mrmeku** [comment]
Test this please

**Comment by thelgevold** [comment]
How can I get access to this function from my external application? I might be misunderstanding this, but I thought the function had to be added to `defs.bzl` for me to be able to import it?

**Comment by mrmeku** [comment]
This was added to rules_nodejs instead
